torch/nn/functional.py

```diff
@@ -2299,6 +2299,13 @@ def pairwise_distance(x1, x2, p=2, eps=1e-6, keepdim=False):
     return torch.pairwise_distance(x1, x2, p, eps, keepdim)
 
 
+def pdist(x, p=2):
+    r"""
+    See :func:`torch.pdist` for details
+    """
+    return torch.pdist(x, p)
```

Comment:
If it's a direct pass through, you should just use `_add_docstr` to attach a docstring to the torch function you're binding. See https://github.com/pytorch/pytorch/blob/88025fe7776fbcfbbb87a0df1f7249348100020b/torch/nn/functional.py#L62 for an example.

Reply:
torch pdist is a native function, i.e. defined in `aten/src/ATen/native/native_functions.yaml`, so it also doesn't have documentation. I'm happy to both add the pass through and documentation, but it's not clear what to do with native functions.

Reply:
I don't see documentation anywhere in the patch?

(Also, what's the plan for `squareform`?)

Reply:
For native functions, the documentation is attached "post-facto" with `_add_docstr`. Check `torch/_tensor_docs.py`

Reply:
`squareform`?

Reply:
Do you mean adding a `squareform` function? It wouldn't be bad, but it probably makes more sense to add something equivalent to numpy's {triu,tril}_indices. They'll allow easy conversion and are a tad more general.